conans/client/tools/settings.py

```diff
@@ -0,0 +1,50 @@
+from conans.errors import ConanInvalidConfiguration
+from conans.client.build.cppstd_flags import cppstd_flag, cppstd_from_settings
+from conans.client.tools.oss import OSInfo
+
+
+def cppstd_minimum_required(conanfile, cppstd, gnu_extensions=False):
+    """ Validate if current cppstd fits the minimal version required.
+
+        In case the current cppstd doesn't fit the minimal version required
+        by cppstd, a ConanInvalidConfiguration exception will be raised.
+
+    :param conanfile: ConanFile instance with cppstd to be compared
+    :param cppstd: Minimal cppstd version required
+    :param gnu_extensions: GNU extension is required (e.g gnu17). This option ONLY works on Linux.
+    """
+    def extract_cpp_version(cppstd):
+        return str(cppstd).replace("gnu", "")
+
+    current_cppstd = cppstd_from_settings(conanfile.settings)
+    if current_cppstd and gnu_extensions and "gnu" not in current_cppstd and OSInfo().is_linux:
+        raise ConanInvalidConfiguration("Current cppstd ({}) does not have GNU extensions, which is"
+                                        " required on Linux platform.".format(current_cppstd))
+    elif current_cppstd and extract_cpp_version(current_cppstd) < extract_cpp_version(cppstd):
```

Comment:
we need to handle C++ 98 too, so this comparison won't work (also, I would cast them to integers)

Reply:
hahaha @lasote just asked to remove the int cast

Reply:
what a bug! 98 > 11